hs_core/models.py

```diff
@@ -1523,7 +1523,8 @@ class Bags(models.Model):
     content_type = models.ForeignKey(ContentType)
 
     content_object = generic.GenericForeignKey('content_type', 'object_id')
-    bag = models.FileField(upload_to='bags', max_length=500, storage=IrodsStorage() if getattr(settings,'USE_IRODS', False) else DefaultStorage(), null=True) # actually never null
+    #bag = models.FileField(upload_to='bags', max_length=500, storage=IrodsStorage() if getattr(settings,'USE_IRODS', False) else DefaultStorage(), null=True) # actually never null
+    bag_url = models.URLField(null=True, blank=True)
```

Comment:
Brian,

I cannot add a note from github directly after I pushed new code change to the feature branch to address this issue, so replying directly here. I have removed bag_url URLField from Bags class in models.py, and generate bag_url dynamically as part of page context variables for resource landing page. It works well based on my testing. Could you do a quick review and see whether everything looks good now so that it can be merged?

Thanks,
Hong

From: Brian Miles [mailto:notifications@github.com]
Sent: Tuesday, May 26, 2015 10:47 AM
To: hydroshare/hydroshare
Cc: Hong Yi
Subject: Re: [hydroshare] 352 new bagit workflow (#402)


In hs_core/models.py<https://github.com/hydroshare/hydroshare/pull/402#discussion_r31040749>:

> @@ -1523,7 +1523,8 @@ class Bags(models.Model):

>      content_type = models.ForeignKey(ContentType)

>

>      content_object = generic.GenericForeignKey('content_type', 'object_id')

> -    bag = models.FileField(upload_to='bags', max_length=500, storage=IrodsStorage() if getattr(settings,'USE_IRODS', False) else DefaultStorage(), null=True) # actually never null

> +    #bag = models.FileField(upload_to='bags', max_length=500, storage=IrodsStorage() if getattr(settings,'USE_IRODS', False) else DefaultStorage(), null=True) # actually never null

> +    bag_url = models.URLField(null=True, blank=True)

My main concern was that the absolute URL not be stored in the database, so it is good you are not storing that. However, I don't think we should be storing even the relative URL. The problem is, if we decide we need to edit urls.py in django_irods changing "download" to something else, then the database contains broken URLs that we then have to deal with some how. Even these relative URLs should be generated dynamically at run time, for example using reverse<https://docs.djangoproject.com/en/1.7/topics/http/urls/#reversing-namespaced-urls>.

Also, I suggest changing django_irods/urls.py so that the:

url(r'^download/$', 'django_irods.views.download'),

is instead:

url(r'^download/(?P<path>[^/]+)$', 'django_irods.views.download'),

then when you reverse do something like:

reverse('django_irods:django_irods.views.download', kwargs={'path': 'BAGNAME'})

Let me know if I can help with any of this.

—
Reply to this email directly or view it on GitHub<https://github.com/hydroshare/hydroshare/pull/402/files#r31040749>.